### PR TITLE
[feat]: [pl-31867]: Adds list permission to be used by list serviceaccount apis

### PR DESCRIPTION
**Pull request by ashishsanodia-harness**
## Describe your changes
Adds list permission to be used by list serviceaccount apis

Tests:
- Verified entry for this new list permissions in access control database 
- Verified entry for roles having this new list permissions in access control database 

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45099)

### CODE REVIEW

**Review by Jimit09Gandhi — APPROVED** [review]
> Reviewed 4 of 5 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45099)* status: :shipit: complete! ✓

File: access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml [hunk 1]
```diff
@@ -446,6 +446,14 @@ permissions:
       - account
       - organization
       - project
+  - identifier: core_serviceaccount_list
+    name: List ServiceAccounts
```
Comment (meenakshi-harness) [inline access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml:450]: space between Service and Accounts

File: access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml [hunk 2]
```diff
@@ -819,6 +825,7 @@ roles:
       permissions:
         - core_license_view
         - core_account_view
+        - core_serviceaccount_list
```
Comment (meenakshi-harness) [inline access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml:828]: Why are we making this permission as part of basic roles?
For now we want to enable this for all customers by default but later also give flexibility to them to enable/disable this.

**Comment by ashishsanodia-harness** [comment]
*[`access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml` line 450 at r2](https://reviewable.io/reviews/harness/harness-core/45099#-NQUFSCEggs9jn2X2pt8-r2-450:-NQUpFR6514g6rO5QtmF:b-886gsx) ([raw file](https://github.com/harness/harness-core/blob/7ece86d3c20a213936454ce017e5fa2308b14bbb/access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml#L450)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

space between Service and Accounts
</blockquote></details>

Fixed

**Comment by ashishsanodia-harness** [comment]
*[`access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml` line 828 at r2](https://reviewable.io/reviews/harness/harness-core/45099#-NQUFSCEggs9jn2X2pt9-r2-828:-NQUpIsw2r4ds7PuUKgD:b-886gsx) ([raw file](https://github.com/harness/harness-core/blob/7ece86d3c20a213936454ce017e5fa2308b14bbb/access-control/service/src/main/resources/io/harness/accesscontrol/roles/managed-roles.yml#L828)):*
<details><summary><i>Previously, meenakshi-harness (Meenakshi Raikwar) wrote…</i></summary><blockquote>

Why are we making this permission as part of basic roles?
For now we want to enable this for all customers by default but later also give flexibility to them to enable/disable this. 
</blockquote></details>

Fixed

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> Reviewed 2 of 5 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45099#-:-NQV0c_ZA6xITL__mAs4:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**